module/zfs/btree.c

```diff
@@ -2064,27 +2113,24 @@ static void
 zfs_btree_verify_poison_helper(zfs_btree_t *tree, zfs_btree_hdr_t *hdr)
 {
 	size_t size = tree->bt_elem_size;
-	if (!hdr->bth_core) {
+	if (!zfs_btree_is_core(hdr)) {
 		zfs_btree_leaf_t *leaf = (zfs_btree_leaf_t *)hdr;
-		uint8_t val = 0x0f;
-		for (int i = hdr->bth_count * size; i < BTREE_LEAF_SIZE -
-		    sizeof (zfs_btree_hdr_t); i++) {
-			VERIFY3U(leaf->btl_elems[i], ==, val);
-		}
+		for (size_t i = (hdr->bth_first + hdr->bth_count) * size;
```

Comment:
Should we also poison/verify the poison of the bytes before bth_first?

Reply:
I do actually poison it in zfs_btree_poison_node() and via zfs_btree_poison_node_at() in bt_shrink_leaf().  I may add verify here and see whether it trigger.

Reply:
It passed the scrub successfully.  Added.